src/gitflow/ui/GitflowOpenTaskPanel.java

```diff
@@ -100,19 +106,34 @@ public void commit() {
         final GitflowBranchUtil.ComboEntry selectedFeatureBaseBranch = (GitflowBranchUtil.ComboEntry) featureBaseBranch.getModel().getSelectedItem();
         final GitflowBranchUtil.ComboEntry selectedHotfixBaseBranch = (GitflowBranchUtil.ComboEntry) hotfixBaseBranch.getModel().getSelectedItem();
 
-        GitflowAction action;
-
         if (startFeatureRadioButton.isSelected()) {
-            action = new StartFeatureAction(myRepo);
-            action.runAction(myProject, selectedFeatureBaseBranch.getBranchName(), featureName.getText());
-            gitflowState.setTaskBranch(currentTask, GitflowConfigUtil.getFeaturePrefix(myProject, myRepo) + featureName.getText());
+            final String branchName = GitflowConfigUtil.getFeaturePrefix(myProject, myRepo) + featureName.getText();
+            runAction(new StartFeatureAction(myRepo), selectedFeatureBaseBranch.getBranchName(), branchName);
         }
         else if (startHotfixRadioButton.isSelected()) {
-            action =  new StartHotfixAction(myRepo);
-            action.runAction(myProject, selectedHotfixBaseBranch.getBranchName(), hotfixName.getText());
-            gitflowState.setTaskBranch(currentTask, GitflowConfigUtil.getHotfixPrefix(myProject, myRepo) + hotfixName.getText());
+            final String branchName = GitflowConfigUtil.getHotfixPrefix(myProject, myRepo) + hotfixName.getText();
+            runAction(new StartHotfixAction(myRepo), selectedHotfixBaseBranch.getBranchName(), branchName);
+        }
+    }
+
+    private void runAction(GitflowAction action, String baseBranchName, final String branchName) {
```

Comment:
I would call this wrapper something like `attachTaskAndRunAction`. Verbose, I know, but I want to differentiate this `runAction` from the `GitflowAction`'s  `runAction` so it's not as confusing